MetaMask/metamask-extension
# Updating Text component to use TS Box

## Explanation

Currently, the `Text` component is using the deprecated JavaScript version of the `Box` component. This blocks the TypeScript migration of the extension AND causes issues with the polymorphic `as` prop. This PR updates the `Text` component to use the TypeScript version of the `Box` and moves the old version of `Text` to `text/deprecated` folder and updates all imports across the extension.

### Why not just update the `Text` component with `Box` if the API is the same?
I have [another PR open doing just that](https://github.com/MetaMask/metamask-extension/pull/19572) but because of undiagnosable e2e test failures, I have resorted to this route. I will then subsequently update all import paths folder by folder to point to the updated component. This has proven to be a successful method in the past(thanks @brad-decker and @DDDDDanica 😁). It reduces the impact on engineers and merge conflicts and will help get to the bottom of the e2e test failures if they exist or if it's just an annoying timeout.

* Fixes #19569

This PR also unblocks [**9 contributor PRs migrating components to TypeScript**](https://github.com/MetaMask/metamask-extension/pulls?q=is%3Aopen+is%3Apr+label%3A%22design-system%22+Migrate)

## Screenshots/Screencaps

### Before

`Text` component in `develop` branch in storybook

https://github.com/MetaMask/metamask-extension/assets/8112138/84ce4a80-80c2-4b6c-92d7-8ebe38748931

Deprecation message for deprecated `Text` component 

![Screenshot 2023-07-12 at 12 31 04 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/f6168566-ae68-4825-b93a-2bc9b06c4ef8)


### After

Showing updated `Text` component in storybook using the TS version of the `Box` component. Showing no visual regressions and all stories and controls are working as expected. I have made some small updates to more easily find the  replace `Typography` with `Text` section and linked the issue


https://github.com/MetaMask/metamask-extension/assets/8112138/b58ec3ef-eb33-4dcd-9475-a38cea61e6f3

## Manual Testing Steps

To check storybook
- Go to the latest build of storybook in this PR 
- Search `Text` in the search panel
- Check the controls, stories and docs are working as expected.

To check all import paths have been updated correctly. 
- The classNames of the 2 `Text` components are slightly different one uses the TypeScript `Box` and has the className `mm-box` vs the JavaScript version `box-`. SO this means that any snapshot tests will pick this up. There should be NO snapshot test updates with this update in this PR

I also did a search for `<Text` and part of the new import path `text/deprecated` to compare the number of files
`165` files vs `164` files for `.js` files with the 1 difference being a comment in a deprecation message

![Screenshot 2023-07-12 at 1 10 13 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/b8d1210f-f161-4c56-8a94-9cfa6d180f06)
![Screenshot 2023-07-12 at 1 10 19 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/2985c958-43b3-4de7-9c86-afa316e360ee)
![Screenshot 2023-07-12 at 1 10 09 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/d25216c3-df6d-4bf7-ac90-d0e113b6c901)

`11` files vs `11` files for `.tsx` files 

![Screenshot 2023-07-12 at 1 12 41 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/b74e9976-d5ec-4fc2-83fb-be8e83a745fb)
![Screenshot 2023-07-12 at 1 12 47 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/8a666539-c001-44a8-963d-d428adab478a)


## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
